gatling-core/src/main/scala/io/gatling/core/action/StopInjector.scala

```diff
@@ -22,13 +22,18 @@ import io.gatling.core.util.NameGen
 
 import akka.actor.ActorRef
 
-class StopInjector(message: Expression[String], controller: ActorRef, val next: Action) extends ChainableAction with NameGen {
+class StopInjector(message: Expression[String], condition: Expression[Boolean], controller: ActorRef, val next: Action) extends ChainableAction with NameGen {
   override val name: String = genName("stopInjector")
 
   override def execute(session: Session): Unit = recover(session) {
-    message(session).map { msg =>
-      logger.error(s"Requested injector to stop: $msg")
-      controller ! ControllerCommand.StopInjector
+    for {
+      condition <- condition(session)
+      msg <- message(session)
+    } yield {
+      if (condition) {
+        logger.error(s"Requested injector to stop: $msg")
+        controller ! ControllerCommand.StopInjector
+      }
       next ! session
```

Comment:
move in else block: no need to propagate the user if we're going to interrupt anyway